### ISSUE TITLE
Migration::dropIndex doesn't able to drop "text" or "hashed" indexes

### ISSUE THREAD

**Issue by askobara** [yiisoft/yii2-mongodb#247]
### What steps will reproduce the problem?
* create migration, for example
```php
<?php
class m180122_141924_add_index_on_message_to_logs extends \yii\mongodb\Migration
{
    public function up()
    {
        $this->createIndex('logs', ['message' => 'text']);
    }

    public function down()
    {
        $this->dropIndex('logs', ['message' => 'text']);
    }
}
```
* apply it
* try to revert changes

### What's expected?
Index have to be deleted

### What do you get instead?
```bash
Error: Index to be dropped does not exist.
```
### Additional info

| Q                      | A
| ---------------------- | ---
| Yii version            | 2.0.13
| Yii MongoDB version    | 2.1.5
| MongoDB server version | 3.6.2
| PHP version            | 7.0.22
| Operating system       | ubuntu 16.10

**Comment by klimov-paul** [comment]
This is incorrect usage:

> $this->createIndex('logs', ['message' => 'text']);

So it is not surprising it does not work. Update your code according the documented specification:
https://github.com/yiisoft/yii2-mongodb/blob/master/Migration.php#L116-L122

**Comment by askobara** [comment]
I'm not sure that you've got my issue.
The index was created successfully, but it couldn't be dropped. Current implementation of the `createIndex` allows to use `'text'` as a type of index. In it's turn, `dropIndex` expect that the index was created only with numeric types.

**Comment by klimov-paul** [comment]
Resolved by commit a6d68fc8edb3d834bcdfc0eac95235f6a5822de0
